indico/util/i18n.py

```diff
@@ -50,12 +50,48 @@ def get_translation_domain(plugin_name=_use_context):
             return get_domain()
 
 
+def __gettext(tr, message):
+    """Look for a translation in both the standard and plural translations.
+
+    Same as gettext.GNUTranslations.gettext but when the translation is not found,
+    it looks for a translation in plurals to find the singular case.
+    When neither is found, the gettext fallback is used.
+
+    This is needed because by default, gettext() only looks through non-pluralized translations.
+    For example, the following cannot be translated with gettext:
+
+    # Polish
+    msgid "Convener"
+    msgid_plural "Conveners"
+    msgstr[0] "Lider"
+
+    To get a translation for 'Convener', we need ngettext('Convener', 'Conveners', 1).
+
+    See GNUTranslations.gettext & ngettext for more details.
+    """
+    missing = object()
+    tmsg = tr._catalog.get(message, missing)
+    if tmsg is missing:
+        try:
+            # Look for message in plural translations
+            tmsg = tr._catalog[(message, tr.plural(1))]
+        except KeyError:
+            if tr._fallback:
+                return tr._fallback.gettext(message)
+            return message
+    return tmsg
+
+
 def _indico_gettext(*args, **kwargs):
     func_name = kwargs.pop('func_name', 'gettext')
     plugin_name = kwargs.pop('plugin_name', None)
 
     translations = get_translation_domain(plugin_name).get_translations()
-    return getattr(translations, func_name)(*args, **kwargs)
+
+    if func_name == 'gettext':
+        return __gettext(translations, *args, **kwargs)
```

Comment:
Don't we need the special logic for `pgettext` as well (gettext w/ context)?

Reply:
yup, added